src/solvers/kriging.jl

```diff
@@ -98,34 +99,28 @@ function preprocess(problem::EstimationProblem, solver::Kriging)
 
       # determine path and neighborhood search method
       if varparams.maxneighbors ≠ nothing
-        # locations with data for variable
-        varlocs = collect(keys(datamap(problem, var)))
-
         if varparams.neighborhood ≠ nothing
           # local search with a neighborhood
           neigh = varparams.neighborhood
 
-          # create a path from the data and outwards
-          # use at most 10^2 points to generate path
-          N = length(varlocs); M = ceil(Int, N/10^2)
-          path = SourcePath(view(varlocs,1:M:N))
-
-          searcher  = NeighborhoodSearcher(pdomain, neigh)
-          bsearcher = BoundedSearcher(searcher, maxneighbors)
+          if neigh isa BallNeighborhood
+            bsearcher = KBallSearcher(pdata, maxneighbors, neigh)
+          else
+            searcher  = NeighborhoodSearcher(pdata, neigh)
+            bsearcher = BoundedSearcher(searcher, maxneighbors)
+          end
         else
           # nearest neighbor search with a distance
           distance = varparams.distance
-          path = LinearPath()
-          bsearcher = NearestNeighborSearcher(pdomain, maxneighbors,
-                                              locations=varlocs, metric=distance)
+          bsearcher = NearestNeighborSearcher(pdata, maxneighbors, metric=distance)
         end
       else
         # use all data points as neighbors
-        path = LinearPath()
         bsearcher = nothing
       end
 
       # save preprocessed input
+      path = LinearPath()
```

Comment:
Given that we always use a linear path, maybe we can erase it from the `preprocess` function entirely.

Reply:
@exepulveda could you please check this one? I think we can just get rid of path if we are always using a linear path. In estimation loop we can just write `traverse(pdomain, LinearPath())`